parse_cmd_test.go

```diff
@@ -1,151 +1,134 @@
+
+	"get-reviews 0":  {"get-reviews", []string{"0"}},
+	"get-reviews 1":  {"get-reviews", []string{"1"}},
+	"get-reviews 5":  {"get-reviews", []string{"5"}},
+	"get-reviews 10": {"get-reviews", []string{"10"}},
+
+	// Commands are case-insensitive.
+	"Help":      {"help", nil},
+	"hElP":      {"help", nil},
+	"sUbScRiBe": {"subscribe", nil},
+
+	// Day names (as keywords) are also case-insensitive
+	"schedule MoN WED fRi": {"schedule", []string{"monday", "wednesday", "friday"}},
+
+	// Review content *is* case-sensitive.
+	"add-review   I :heart: Pairing Bot!\n": {"add-review", []string{"I :heart: Pairing Bot!"}},
 }
 
-var tableWithArgs = []struct {
-	testName   string
-	inputStr   string
-	wantedCmd  string
-	wantedArgs []string
-	expectErr  bool
-}{
-	{"schedule_1_arg", "schedule monday", "schedule", []string{"monday"}, false},
-	{"schedule_2_args", "schedule monday friday", "schedule", []string{"monday", "friday"}, false},
-	{"schedule_3_args", "schedule monday wednesday friday", "schedule", []string{"monday", "wednesday", "friday"}, false},
-	{"schedule_4_args", "schedule monday wednesday friday sunday", "schedule", []string{"monday", "wednesday", "friday", "sunday"}, false},
-	{"schedule_weekend_only", "schedule sunday", "schedule", []string{"sunday"}, false},
-	{"schedule_wrong_usage", "schedule", "help", nil, true},
-	{"skip_correct_usage", "skip tomorrow", "skip", []string{"tomorrow"}, false},
-	{"skip_wrong_usage", "skip monday", "help", nil, true},
-	{"skip_wrong_usage", "skip whenever", "help", nil, true},
-	{"skip_wrong_usage", "skip", "help", nil, true},
-	{"unskip_correct_usage", "unskip tomorrow", "unskip", []string{"tomorrow"}, false},
-	{"unskip_wrong_usage", "unskip today", "help", nil, true},
-	{"unskip_wrong_usage", "unskip friday", "help", nil, true},
-	{"unskip_wrong_usage", "unskip", "help", nil, true},
-	{"version_extra_args_ok", "version info", "version", nil, false},
+var rejectedCommands = []string{
+	"",
+
+	// Funnily enough, these *do* give you what you want!
+	"help me",
+	"halp",
+	"schedule help",
+
+	// Unexpected arguments
+	"status me",
+	"cookie me",
+
+	// Did they really want `schedule`?
+	"subscribe tue",
+	"unsubscribe thu",
+
+	// (Un)skipping requires an argument.
+	"skip",
+	"unskip",
+
+	// TODO(#49): Allow (un)skipping days other than tomorrow
+	"skip friday",
+	"unskip next",
+
+	// This is not the way to delete reviews you don't like 😛
+	"get-reviews -1",
+	"get-reviews -10",
+
+	// Unknown commands
+	"scheduleing monday",
+	"schedul monday",
+	"mooh",
 }
 
-func TestParseCmdWithArgs(t *testing.T) {
-	for _, tt := range tableWithArgs {
-		t.Run(tt.testName, func(t *testing.T) {
-			gotCmd, gotArgs, gotErr := parseCmd(tt.inputStr)
-			if gotCmd != tt.wantedCmd || len(gotArgs) != len(tt.wantedArgs) {
-				t.Errorf("got %v, %v, wanted %v, %v\n", gotCmd, gotArgs, tt.wantedCmd, tt.wantedArgs)
+func Test_parseCmd(t *testing.T) {
+	for input, expected := range acceptedCommands {
```

Comment:
I'd usually do "one top-level test per table" - `TestParseCmdAccepts`, `TestParseCmdRejects`

But this is also fine, the error condition should trace back to the right point.

Reply:
I had considered doing that at least once while writing this, but didn't only out of laziness. So this is the nudge I needed to go do that :smile: